app/assets/javascripts/live_search.js

```diff
@@ -118,6 +126,65 @@
     return $.param(this.state) !== this.$form.serialize();
   };
 
+  LiveSearch.prototype.updateOrder = function updateOrder() {
+    if (!this.$orderSelect.length) {
+      return
+    }
+
+    var liveSearch = this;
+
+    var keywords = this.getTextInputValue('keywords', this.state);
+    var previousKeywords = this.getTextInputValue('keywords', this.previousState);
+
+    var keywordsPresent = keywords !== "";
+    var keywordsBlank = !keywordsPresent;
+
+    var previousKeywordsPresent = previousKeywords !== "";
+    var previousKeywordsBlank = !previousKeywordsPresent;
+
+    var keywordsChanged = keywordsPresent && (previousKeywordsBlank || (keywords !== previousKeywords));
+    var keywordsCleared = keywordsBlank && previousKeywordsPresent;
+
+    if (keywordsPresent) {
+      liveSearch.insertRelevanceOption();
+    } else {
+      liveSearch.removeRelevanceOption();
+    }
+
+    if (keywordsCleared) {
+      liveSearch.selectDefaultSortOption();
+    }
+
+    if (keywordsChanged) {
+      liveSearch.selectRelevanceSortOption();
+    }
+  };
+
+  LiveSearch.prototype.selectDefaultSortOption = function selectDefaultSortOption() {
+    var defaultSortOption = this.$orderSelect.data('default-sort-option');
+
+    this.$orderSelect.val(defaultSortOption);
+    this.state = this.$form.serializeArray();
+  };
+
+  LiveSearch.prototype.selectRelevanceSortOption = function selectRelevanceSortOption() {
+    var relevanceSortOption = this.$orderSelect.data('relevance-sort-option');
+
+    this.$orderSelect.val(relevanceSortOption);
+    this.state = this.$form.serializeArray();
+  };
+
+  LiveSearch.prototype.insertRelevanceOption = function insertRelevanceOption() {
+    var adjacentOption = this.$orderSelect.children("option").eq(this.$relevanceOrderOptionIndex);
+
+    adjacentOption.before(this.$relevanceOrderOption);
+  };
+
+  LiveSearch.prototype.removeRelevanceOption = function removeRelevanceOption() {
+    this.$relevanceOrderOption.removeAttr('disabled');
+    this.$relevanceOrderOption.remove();
```

Comment:
Is it necessary to remove an attribute from the element before removing it?

Reply:
The server side render will disable the relevance option so we either need to remove it here, or when we add it back, or maybe when we initialise the live search.

Reply:
Ah I see, thanks